src/shape.rs

```diff
@@ -242,6 +242,13 @@ impl Shape {
         })
     }
 
+    pub(crate) fn add_width(&self, width: usize) -> Shape {
+        Shape {
+            width: self.width + width,
+            ..*self
+        }
+    }
```

Comment:
Consider this case:

```
// With max width                     40
//                                     |
use long_segment_loooooooooooooooooong::{Foo, Bar};

// should be formated to
use long_segment_loooooooooooooooooong::{
    Foo,
    Bar,
};
```

Look at the `UseTree::rewrite()`, especially

```
let s = if prev_is_allow_overflow
    && matches!(segment.kind, UseSegmentKind::List(_))
{
    shape.add_width(1)
} else {
    shape.clone()
};
```

Let's trace the behavior of the format above.

```
committed = "use "

// 35 = 40 - (4 + 1)
shape = {indent: 0, offset: 4, width: 35}
// len = 34
segment = "long_segment_loooooooooooooooooong"

// 34 + 2 exceeds 35, but the above segment should be stacked to the current line because this is the first segment
committed = "use long_segment_loooooooooooooooooong::"
// with overflow.
shape = {indent: 0, offset: 40, width: 0}
prev_is_allow_overflow = true

// TIMING1
segment = "{Foo, Bar}"
// Here, we must secure the room for "{" since `prev_is_allow_overflow == true`, or `segment.rewrite(context, shape)?` fails.
s = {indent: 0, offset: 40, width: 1}

committed = "use long_segment_loooooooooooooooooong::{\n    Foo,\n    Bar,\n}"
```

Alternative options:

1. Use `segment.rewrite(context, shape.infinite_width())?;` at the `TIMING1` instead.

We can't do that because it produces

```
use long_segment_loooooooooooooooooong::{Foo, Bar};
```

2. Add a variant of `UseSegment::rewirte()` that can handle yet another argument `should_commit_open_brace_even_if_no_width`.

It's not simple compared to adding `Shape::add_width()`...

So, I added `Shape::add_width()`.